### PR TITLE
ASoC: rt700/711/1308-sdw: patches update

### PR DESCRIPTION
**Pull request by shumingfan**
There are some modification for rt700/rt711/rt1308-sdw codec driver.

### CODE REVIEW

File: sound/soc/codecs/rt711-sdw.c [hunk 1]
```diff
@@ -144,6 +144,9 @@ static int rt711_update_status(struct sdw_slave *slave,
 	/* Update the status */
 	rt711->status = status;
```
Comment (dbaluta) [inline sound/soc/codecs/rt711-sdw.c:81]: @shumingfan Please make one patch per logical change and for each patch please explain the reason why the patch is needed. 

Also, please use git log  sound/soc/codecs/rt711-sdw.c and notice the style of the commit subject.

So, in your case you will have 3 patches:

1) ASoC: codecs: rt711: Simplify return value

ret is always 0 and never changed so directly return 0 in order to remove one variable
from stack and also make code more readable.

``2) and 3)`` in the same manner.

**Comment by dbaluta** [comment]
General comment: please make your commit messages more specific. 'Minor change' can mean a lot of things :)

File: sound/soc/codecs/rt700.c [hunk 2]
```diff
@@ -77,7 +77,7 @@ static int rt700_index_write(struct regmap *regmap,
 static int rt700_index_read(struct regmap *regmap,
 			    unsigned int reg, unsigned int *value)
 {
-	int ret;
+	int ret = 0;
```
Comment (plbossart) [inline sound/soc/codecs/rt700.c]: this init is not required, you have a ret = regmap_write(regmap, RT700_PRIV_DATA_R_H, val_h);

File: sound/soc/codecs/rt700.c [hunk 3]
```diff
@@ -256,6 +255,10 @@ static void rt700_jack_detect_handler(struct work_struct *work)
 			SND_JACK_HEADSET |
 			SND_JACK_BTN_0 | SND_JACK_BTN_1 |
 			SND_JACK_BTN_2 | SND_JACK_BTN_3);
+	snd_soc_jack_report(rt700->hs_jack, rt700->jack_type,
+			SND_JACK_HEADSET |
+			SND_JACK_BTN_0 | SND_JACK_BTN_1 |
+			SND_JACK_BTN_2 | SND_JACK_BTN_3);
```
Comment (plbossart) [inline sound/soc/codecs/rt700.c:264]: for my education, why do we call snd_soc_jack_report() twice? I don't get what the code does here, maybe a comment would help?

File: sound/soc/codecs/rt700.c [hunk 4]
```diff
@@ -307,6 +310,10 @@ static void rt700_btn_check_handler(struct work_struct *work)
...
```
Comment (plbossart) [inline sound/soc/codecs/rt700.c:321]: same here, what does this do?

File: sound/soc/codecs/rt700.c (hunk 2 above)
Comment (shumingfan) [reply sound/soc/codecs/rt700.c]: Ok, got it.

File: sound/soc/codecs/rt700.c (hunk 4 above)
Comment (shumingfan) [reply sound/soc/codecs/rt700.c:321]: I just want the button response like click one by one.
If a user keeps pressing the button, the system will report for volume+/- too quickly.
If this modification is not well, another way I could adjust the delay time of the work queue.

Comment (plbossart) [reply sound/soc/codecs/rt700.c:321]: I see. The issue I have is that not all buttons are mapped to volume...
How is this solved with other codecs, it doesn't look like an rt700 issue only, does it?

Comment (shumingfan) [reply sound/soc/codecs/rt700.c:321]: Yeah, I checked the source code how to do in HDA mode.
The snd_hda_jack_report_sync function in sound/pci/hda/hda_jack.c reports twice if it has the button_state.

Comment (plbossart) [reply sound/soc/codecs/rt700.c:321]: did you mean the code below
````c
state = jack->button_state;
			if (get_jack_plug_state(jack->pin_sense))
				state |= jack->type;
			snd_jack_report(jack->jack, state);
			if (jack->button_state) {
				snd_jack_report(jack->jack,
						state & ~jack->button_state);
				jack->button_state = 0; /* button released */
			}
````

If yes, then you should add a comment that this deals with button released or explain why you have two calls which seem to do the same thing.

Comment (shumingfan) [reply sound/soc/codecs/rt700.c:321]: Yes, I will make a patch for button issue and resend the patches.

File: sound/soc/codecs/rt1308-sdw.c [hunk 5]
```diff
@@ -30,36 +30,13 @@
...
-	case 0x2012 ... 0x2016:
-	case 0x201a ... 0x2027:
-	case 0x2029 ... 0x202a:
-	case 0x202d ... 0x2034:
-	case 0x2100 ... 0x213f:
-	case 0x2200 ... 0x2204:
-	case 0x2206 ... 0x2212:
-	case 0x2220 ... 0x2223:
-	case 0x2230 ... 0x2231:
+	case 0x2f01 ... 0x2f07:
```
Comment (plbossart) [inline sound/soc/codecs/rt1308-sdw.c:35]: are the 0xe0 and 0xf0 intentional?

Comment (shumingfan) [reply sound/soc/codecs/rt1308-sdw.c:35]: @plbossart The driver will access 0xe0/0xf0 registers in rtXXX_clock_config function. So, I keep it in readable_register function.

Comment (plbossart) [reply sound/soc/codecs/rt1308-sdw.c:35]: ok, thanks for confirming.

**Pull request merged**